content/articles/introduction-to-clean-architecture-and-asp-dotnet-core-implementation/index.md

```diff
@@ -0,0 +1,456 @@
+
+3. Maintainability is optimized using clean architecture.
+
+Third, a clear architecture makes it easier to maintain. We spend more time and money maintaining a typical corporate program with a long life cycle, say 12 years or more, than we do design it. So, if we optimize for maintainability, as clean architecture does, we should be able to lower the cost of software maintenance.
+
+The advantages of clean architecture include;
+
+- External agency independence. In reality, your business rules have no idea of what is going on in the outside world.
+
+- User interface independence. Changes to the user interface can be made without affecting the rest of the system or business rules.
+
+- Testable. It can now be tested on its own. You can test business rules without taking into account the user interface, database, or mock servers.
+
+- Independence of Database and Frameworks. Because the database and object-relation mapping (ORM) can be quickly changed, the software will not be reliant on them.
+
+
+### Clean Architecture implementation with ASP dotNET core.
+
+
+After learning about this architecture in general, we can now learn how to implement the clean architecture in the ASP.NET core application model.
+
+Having said this, we shall go into our visual studio, where will now generate a blank solution.
+
+![The digram shows how to configure the new project](engineering-education/introduction-to-clean-architecture-and-implementation-with-asp.net-core/blanksolution.png)
+
+After that, we will add a Web API project to our solution, along with three class library projects.
+This is how the project will appear.
+
+![The digram shows how our project will look like after adding some classes](engineering-education/introduction-to-clean-architecture-and-implementation-with-asp.net-core/addingclass.jpg)
+
+We now add the project references. The key concept of clean architecture is that the inner layer is not supposed to know the outer layer while the outer layer should know the inner layer. Therefore core will not have project reference but infrastructure will know about core and API will know about all three layers.
+
+We can now go to the core project and create two folders named Entities and the other *Repositories*. Withing our *entities* file, we can then have to create a class named *employee. cs.*
+
+Below is an example of a code. 
+
+```C#
+
+public class Employees {
+    [Key]
+    [DbGenerate(DbGenerateOption.Identity)]
+    public Int32 EmployeesId {
+        get;
+        set;
+    }
+    public string FirstName {
+        get;
+        set;
+    }
+    public string LastName {
+        get;
+        set;
+    }
+    public string PhoneNumber {
+        get;
+        set;
+    }
+    public string Email {
+        get;
+        set;
+    }
+}
+
+```

```

Comment:
I am finding trouble understanding what this piece of code is supposed to be doing and why we are adding it to the project. Kindly clarify what the end goal with this piece of code is.